contracts/stake/src/state.rs

```diff
@@ -118,15 +132,15 @@ impl StakeState {
             .expect("A stake should exist in the map to be unstaked!");
         let prev_stake = Some(*loaded_stake);
 
-        // ensure there is a value staked, and that the withdrawal is exactly
-        // the same amount
+        // ensure there is a value staked, and that the withdrawal is not
+        // greater than the available funds
         let stake = loaded_stake
             .amount
-            .as_ref()
+            .as_mut()
             .expect("There must be an amount to unstake");
 
-        if value != stake.total_funds() {
-            panic!("Value withdrawn different from staked amount");
+        if value > stake.total_funds() {
+            panic!("Value to unstake higher than the staked amount");
```

Comment:
The check for the real eligibility is done by the consumer of the stake contract (consensus).

Reply:
On a second though, I believe that we should introduce the proposed check.

At least to prevent people to pollute the stake contract with very low staked amount

Reply:
The check is in the staking-contract when staking so I would say either not check in the contract or also add the check when partially unstaking.

Reply:
Yes, I agree that we need in term of consistency. Going to change it

Reply:
Check added